### ISSUE TITLE
training result

### ISSUE THREAD

**Issue by liuxm117** [aqlaboratory/openfold#108]
hi, I trained openfold and use the ckpt to test sequence, but it was not correct, lots of atoms overlapped together
![Uploading 微信图片_20220525170716.jpg…]()

**Comment by liuxm117** [comment]
![微信图片_20220525170716](https://user-images.githubusercontent.com/50068070/170226479-6d0a8989-648e-4c17-a099-6a62d24449a1.jpg)

**Comment by gahdritz** [comment]
It's difficult to say what might be happening here. OpenFold is designed for proteins, so training it on small molecules like this one requires extensive modifications, I imagine. All I can say is that our own trained version of OpenFold has matched AlphaFold's performance on proteins.

**Comment by liuxm117** [comment]
Congratulations， hope to see your trained weights soon,
1、what precision are you using，use fp16, loss is always nan
2、In the later stage of training, the loss has been oscillating and no longer decreasing，can you give me some advice，which part or which parameter should I pay attention to

**Comment by gahdritz** [comment]
1. Yes, fp16 is unstable to the point where I wouldn't currently call it functional. CASP + our own training (in bfloat16) is currently eating up most of my development time, but this remains one of the top features I'd like to get working.
2. It's almost impossible to say without knowing what data you're using, what your task is, what modifications you've made, etc. In any case, training is expensive enough that I haven't been able to develop much intuition for it---all I know for certain is that DeepMind's presets work pretty well for the protein folding task. Anything beyond that is pretty much terra incognita for me too, I'm afraid.

**Comment by liuxm117** [comment]
thank you for your reply，I use pdb_mmcif as train dataset, I almost didn't modify the code, just did some memory optimization to speed up the performance，maybe it's because I don't have enough gpus

**Comment by gahdritz** [comment]
Oh are you not doing some kind of small molecule conformation prediction? I assumed that you're not doing the standard protein folding task from that image you sent.

**Comment by liuxm117** [comment]
I use pymol to visualization predicted pdb，but it doesn't show anything ,this image is visualization by another tool, I used 24GPUS, Almost 100w samples were trained

**Comment by gahdritz** [comment]
How are the metrics looking? What is your current LDDT-Ca?

**Comment by liuxm117** [comment]
![图片1](https://user-images.githubusercontent.com/50068070/170436594-830d54c4-ac91-44cf-bf01-10b184841851.jpg)

**Comment by gahdritz** [comment]
![image](https://user-images.githubusercontent.com/26167235/170437505-9a607640-049d-4def-98b4-25ddaefb1c3b.png)
Here's our validation LDDT curve. The x axis records batches of size 132, so the plateau at 0.8 LDDT is reached after the model's seen approx. 660k proteins. This behavior has been extremely consistent across our training runs. Perhaps you just need to train longer?

**Comment by liuxm117** [comment]
due to the limitation of the number of GPUs， I cannot increase the length of batch_size，Could this be the problem? Where did your validation set come from?(cameo? or just a part of trainset)

**Comment by gahdritz** [comment]
Yes CAMEO.

**Comment by vetmax7** [comment]
Hello @liuxm117  and all!

Can you explain please, how did you use your `*.ckpt ` files for for prediction? Which script did you use for test with ckpt files?
In Readme is mentioned only about  `*.pt`.

**Comment by vetmax7** [comment]
@gahdritz  @liuxm117   Can you explain please, how did you use your *.ckpt  files for for prediction as a model? Or how can I convert it to *.npz? If I try to convert my ckpt I've got a lot of errors about different structure of my ckpt.
